dts/bindings/modem/quectel,bg96/quectel,bg96.yaml

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2022 Trackunit
+# SPDX-License-Identifier: Apache-2.0
+
+description: Quectel BG96 modem
+
+compatible: "quectel,bg96"
+
+include: [base.yaml, uart-device.yaml]
+
+properties:
+    label:
+      required: true
+
+    power-gpios:
+      type: phandles
+      required: true
+
+    reset-gpios:
+      type: phandles
+      required: false
+
+    powerdown-gpios:
+      type: phandles
+      required: false
+
+    nmea-uart:
+      type: path
+      required: false
+      description: |
+        Used for NMEA frames from GNSS if included,
+        otherwise,the location is read from the modem
+        using AT commands.
+
+    apn-name:
+      type: string
+      required: false
+      description: APN name for network connection context
+
+    username:
+      type: string
+      required: false
+      description: Username for network connection context
+
+    password:
```

Comment:
having a username/password dts feels wrong.

Reply:
It is not possible to configure APN during runtime (at least yet), so it has to be here or in a KConfig, the reason it is here is to allow for multiple modem instances, which could use different APNs, I can imagine having two identical modems on different networks for redundancy for example

Reply:
A username/password may be needed to integrate with existing hardware, but we should design APIs to comply (support?) with each of the provisions in ETSI EN 303645: CYBER; 
Cyber Security for Consumer Internet of Things: 
Baseline Requirements (https://www.etsi.org/deliver/etsi_en/303600_303699/303645/02.01.01_60/en_303645v020101p.pdf)